lib/presentation/auth/widgets/enter_username.dart

```diff
@@ -100,9 +139,8 @@ class _EnterUserNameState extends State<EnterUserName> {
                     onTap: () {
                       if (_isNameValid && state is! AwaitingUsernameUpdate) {
                         FocusScope.of(context).unfocus();
-                        context
-                            .read<AuthBloc>()
-                            .add(AuthEvent.updateUsername(_username!));
+                        context.read<AuthBloc>().add(
+                            AuthEvent.updateUsername(_firstname!, _lastname!));
```

Comment:
You should create a new BLoC as I described:

- Create a new BLoC called UsernameBloc
- It should have 1 event: UpdateUsernameEvent
- It should have 3 states:
  - UpdateFailed - Something went wrong, display error and let them try again
  - UpdateSuccessful - It was succesful, show them the next page
  - Updating - While it's updating show the loading pill button version instead, and make sure clicking it doesn't trigger another event.
- The UsernameBloc should have IProfileRepository, see:
  - final IProfileRepository profileRepository;
- The UsernameBloc should be annotated with @injectable
- When instantiating the UsernameBloc, do it like this:
- `create: (context) => getIt<UsernameBloc>()`

And t hen you should completely remove any reference to AuthBloc in this widget, it should be completely decoupled from the outside. No sending an event to the AuthBloc, no using AuthBloc to build or change state.

Reply:
Implement almost every point mentioned in this, if still remain something pls check